### PR TITLE
Form Components: ES6ify

### PR DESCRIPTION
**Pull request by ockham**
For _Project Hubris_.

To test:
* Verify that all components in http://calypso.localhost:3000/devdocs/design/form-fields still work (watch the console for errors!)
* To be sure, also try a couple of form-laden Calypso sections, e.g. `/me`, or site settings.

The `export` shouldn't cause any trouble since there aren't any named ones (just `default` ones), so any `import`s/`require`s should continue to work.

### CODE REVIEW

File: client/components/forms/form-phone-input/test/index.jsx [hunk 1]
```diff
@@ -1,77 +1,57 @@
...
-	} );
+describe( 'FormPhoneInput', function() {
+	useFakeDom();
 
 	describe( 'getValue()', function() {
 		it( 'should set country from props', function() {
-			var phoneComponent = ReactDom.render( <FormPhoneInput countriesList={ mockCountriesList } initialCountryCode={ countries[ 1 ].code } />, container );
-
-			expect( phoneComponent.getValue().countryData ).to.deep.equal( countries[ 1 ] );
+			const phoneComponent = shallow( <FormPhoneInput countriesList={ mockCountriesList } initialCountryCode={ countries[ 1 ].code } /> );
```
Comment (tyxla) [inline client/components/forms/form-phone-input/test/index.jsx]: Let's split those props on separate lines to avoid the ESLint line length warning.

Comment (tyxla) [inline client/components/forms/form-phone-input/test/index.jsx]: Actually it appears that this was the reason behind the tests failing.

**Comment by ockham** [comment]
> BTW this PR could have been a little smaller, don't you think? 😉 I'm usually hesitant to merge PRs with so many changes, because that makes it more difficult to spot subtle bugs.

Yeah 😕 It was a bit difficult to anticipate how big this would end up when I started running codemods one by one. I had hoped it would be contained enough, but it turned out pretty big. Thanks for reviewing it all the more!

File: client/components/forms/counted-textarea/index.jsx [hunk 2]
```diff
@@ -2,55 +2,54 @@
...
-		acceptableLength: React.PropTypes.number,
-		showRemainingCharacters: React.PropTypes.bool
-	},
+	static defaultProps = {
+		value: '',
+		placeholder: '',
+		countPlaceholderLength: false,
+		onChange: noop,
+		showRemainingCharacters: false,
+		translate,
```
Comment (ockham) [inline client/components/forms/counted-textarea/index.jsx]: A bit dirty, but it does the trick 😄

File: client/components/forms/counted-textarea/test/index.jsx [hunk 3]
```diff
@@ -63,45 +54,50 @@ describe( 'index', function() {
...
+		const value = '';
+		const placeholder = 'placeholder test';
 
-		renderer.render( <CountedTextarea value={ value } className="custom-class" placeholder={ placeholder } countPlaceholderLength={ true } /> );
-		result = renderer.getRenderOutput();
+		renderer.render(
+			<CountedTextarea value={ value }
+				className="custom-class"
+				placeholder={ placeholder }
+				countPlaceholderLength={ true } />
```
Comment (ockham) [inline client/components/forms/counted-textarea/test/index.jsx]: > Also, personally I prefer closing those ` />` on the next line, and aligning them vertically with the opening `<CountedTextarea` - feels much more readable to me. It's your call really, but if you decide to go that way, you should probably update the rest of the PR (I've seen several more of these).

Mhm, so far, I've kept the closing `/>` on the same line as the last prop, but you're making a good point about legibility. Anyway, I'll opt to stick as-is due to PR size 😊

File: client/components/forms/form-buttons-bar/index.jsx [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
  */
-var React = require( 'react' ),
-	classnames = require( 'classnames' ),
-	omit = require( 'lodash/omit' );
+import React from 'react';
+import classnames from 'classnames';
+import { omit } from 'lodash';
 
-module.exports = React.createClass( {
+export default class extends React.Component {
```
Comment (ockham) [inline client/components/forms/form-buttons-bar/index.jsx]: No reason, other than just letting the codemod do its thing 😄

Comment (ockham) [inline client/components/forms/form-buttons-bar/index.jsx]: (I'll convert it tho.)

File: client/components/forms/form-text-input-with-action/index.jsx [hunk 5]
```diff
@@ -45,38 +46,38 @@ export default class FormTextInputWithAction extends Component {
 		};
 	}
 
-	handleFocus = ( e ) => {
+	handleFocus = e => {
```
Comment (ockham) [inline client/components/forms/form-text-input-with-action/index.jsx]: I'll blame it on the codemod. Adding them back.

File: client/components/forms/clipboard-button/index.jsx [hunk 6]
```diff
@@ -1,59 +1,62 @@
...
+	};
 
-	render: function() {
+	render() {
 		var classes = classNames( 'clipboard-button', this.props.className );
 
 		return (
 			<Button
 				ref="button"
 				{ ...omit( this.props, Object.keys( this.constructor.propTypes ) ) }
```
Comment (ockham) [inline client/components/forms/clipboard-button/index.jsx]: Adding to `propTypes`.

File: client/components/forms/form-label/index.jsx [hunk 7]
```diff
@@ -19,14 +22,17 @@ const FormLabel = ( { children, required, translate, className, ...extraProps }
 	}
 
 	return (
-		<label { ...omit( extraProps, 'moment', 'numberFormat' ) } className={ classnames( className, 'form-label' ) } >
+		<label
+			{ ...omit( extraProps, 'moment', 'numberFormat' ) }
```
Comment (ockham) [inline client/components/forms/form-label/index.jsx]: No, that's already `omit`ted from `extraProps` a couple of lines above.

**Pull request merged**